qubes/tools/qubesd.py

```diff
@@ -55,6 +55,22 @@ def main(args=None):
         loop.add_signal_handler(getattr(signal, signame),
             sighandler, loop, signame, servers)
 
+    # Stop storage for domains not currently running
+    async def stop_storage():
+        async def stop(i):
+            async with i.startup_lock:
+                if not i.is_running():
+                    await i.storage.stop()
+        future = (qubes.utils.void_coros_maybe(stop(i) for i in self.domains),)
+        finished = ()
+        while future:
+            qubes.utils.systemd_extend_timeout()
+            finished, future = await asyncio.wait(
+                future, loop=loop, timeout=30000000)
+        for i in finished:
+            await i
+    loop.run_until_complete(stop_storage())
```

Comment:
I meant to keep the `stop_storage()` function in app.py and just call it from qubesd.py.

Reply:
Ah okay!  Easy fix :slightly_smiling_face: